src/main/java/org/folio/dcb/service/impl/LendingLibraryServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.dcb.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.domain.mapper.TransactionMapper;
+import org.folio.dcb.domain.dto.DcbTransaction;
+import org.folio.dcb.domain.dto.TransactionStatus;
+import org.folio.dcb.domain.dto.TransactionStatusResponse;
+import org.folio.dcb.domain.entity.TransactionEntity;
+import org.folio.dcb.exception.ResourceAlreadyExistException;
+import org.folio.dcb.repository.TransactionRepository;
+import org.folio.dcb.service.LibraryService;
+import org.folio.dcb.service.RequestService;
+import org.folio.dcb.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service("lendingLibraryService")
+@RequiredArgsConstructor
+@Log4j2
+public class LendingLibraryServiceImpl implements LibraryService {
+
+  private final UserService userService;
+  private final RequestService requestService;
+  private final TransactionRepository transactionRepository;
+  private final TransactionMapper transactionMapper;
+
+  @Override
+  public TransactionStatusResponse createTransaction(String dcbTransactionId, DcbTransaction dcbTransaction) {
+    log.debug("createTransaction:: creating a new transaction with dcbTransactionId {} , dcbTransaction {}",
+      dcbTransactionId, dcbTransaction);
+
+    checkTransactionExistsAndThrow(dcbTransactionId);
+    var item = dcbTransaction.getItem();
+    var patron = dcbTransaction.getPatron();
+
+    var user = userService.createOrFetchUser(patron);
+    requestService.createPageItemRequest(user, item);
+    saveDcbTransaction(dcbTransactionId, dcbTransaction);
```

Comment:
I am thinking if we should first create the transaction and then create the user and other needed requests ? Does the order matters ?

Reply:
I think that if the request creation part or virtual patron creation part is failed, then we don't need to persist the transaction.

Reply:
I think this method should be annotated with @transactional , why because what if user creation part successful and transaction creation part failed due to some reason . we will have a redundant user.